File: QTD.SPA/src/app/_DtoModels/SchedulesClassses/Rosters/RosterOverviewVM.ts
import { CBT_ScormRegistration } from "../../CBT/CBT_ScormRegistration";

export class RosterOverviewVM {
  employeeName!:string;
  employeeEmail!:string;
  employeeImage!:any;
  pretestStatus!:string;
  cbtStatus!:string;
  testStatus!:string
  retakeCount!:number;
  score!:number;
  grade!:string;
  gradeNotes!:string;
  classEmployeeId!:any;
  isTestReleased!:boolean;
  isPreTestReleased!:boolean;
  isCBTReleased!:boolean;
  evaluationCompletedDate:Date|string;
  classScheduleEmployeeId:string;
  taskQualificationCompleted:boolean;
}

File: QTD2.Infrastructure/Model/ClassSchedule_Roster/RosterOverviewVM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using QTD2.Domain.Entities.Core;

namespace QTD2.Infrastructure.Model.ClassSchedule_Roster
{
    public class RosterOverviewVM
    {
        public int ClassEmployeeId { get; set; }
        
        public string EmployeeName { get; set; }

        public string EmployeeEmail { get; set; }

        public string EmployeeImage { get; set; }

        public string PretestStatus { get; set; }
        
        public string CbtStatus { get; set; }
        
        public string TestStatus { get; set; }
        
        public int RetakeCount { get; set; }

        public int? Score { get; set; }
        
        public string Grade { get; set; }
        
        public string GradeNotes { get; set; }

        public bool? isTestReleased { get; set; }
        public bool? isPreTestReleased { get; set; }
        public bool? isCBTReleased { get; set; }

        public bool CompletedDate { get; set; }
        public DateTime? EvaluationCompletedDate { get; set; }
        public int ClassScheduleEmployeeId { get; set; }
        public bool TaskQualificationCompleted { get; set; }

    }
}
